DS/DsAssignment4/PrintApplication.java

```diff
@@ -0,0 +1,57 @@
+public class PrintApplication {
+
+	private PriorityQueue priorityQueue = new PriorityQueue(20);
+	public boolean printerStatus;
+
+	private boolean isPrinterFree() {
+		return printerStatus;
+
+	}
+
+	public String processRequest(Task task) {
+		String result = "";
+		if (isPrinterFree() && priorityQueue.checkIsEmpty()) {
+			result = print(task);
+
+		} else {
+			priorityQueue.insert(task);
+			System.out.println("you are in queue...plz wait for some time");
+			result = "you are in queue...plz wait for some time";
+		}
+
+		return result;
+
+	}
+
+	public String processQueue() {
```

Comment:
processQueue() method name should not be exposed outside. The better name is "Print", isn't "processQueue()" an internal thing? :)